types/ember__component/index.d.ts

```diff
@@ -106,10 +104,28 @@ export default class Component extends CoreView.extend(
  * @param object the object to associate with the componetn manager
  * @return the same object passed in, now associated with the manager
  */
-export function setComponentManager<T>(
-    managerFactory: (owner: unknown) => ComponentManager<unknown>,
-    object: T
-): T;
+export function setComponentManager<T>(managerFactory: (owner: unknown) => ComponentManager<unknown>, object: T): T;
+
+// In normal TypeScript, these built-in components are essentially opaque tokens
+// that just need to be importable. Declaring them with unique interfaces
+// like this, however, gives tools like Glint (that DO have a richer
+// notion of what they are) a place to install more detailed type information.
+export interface Input extends Opaque<'component:input'> {}
+export interface Textarea extends Opaque<'component:textarea'> {}
+
+/**
+ * The `Input` component lets you create an HTML `<input>` element.
+ *
+ * @see https://api.emberjs.com/ember/4.1/classes/Ember.Templates.components/methods/Input?anchor=Input
+ */
+export const Input: Input;
+
+/**
+ * The `Textarea` component inserts a new instance of `<textarea>` tag into the template.
+ *
+ * @see https://api.emberjs.com/ember/4.1/classes/Ember.Templates.components/methods/Textarea?anchor=Textarea
```

Comment:
The URL says 4.1 but the file header says 4.0 - should it be updated?

Reply:
Good question. That’s the latest version of Ember docs at present but the API is available from 4.0 forward. I think docs are only available later for uninteresting reasons.

Reply:
If the header reflects the API version, that's good enough for me. 😄